src/main/java/me/zeroeightsix/kami/module/modules/movement/AutoWalk.java

```diff
@@ -33,25 +34,63 @@
+            disable();
+        }
+    });
+
+    public void onEnable() {
+        if (normalizeAngle(mc.player.rotationYaw) >= -22.5 && normalizeAngle(mc.player.rotationYaw) <= 22.5) { // +Z
+            BaritoneAPI.getProvider().getPrimaryBaritone().getCustomGoalProcess().setGoalAndPath(new GoalXZ((int) mc.player.posX, (int) mc.player.posZ + Integer.MAX_VALUE));
+
+            direction = "+Z";
+        } else if (normalizeAngle(mc.player.rotationYaw) >= 22.6 && normalizeAngle(mc.player.rotationYaw) <= 67.5) { // -X / +Z
+            BaritoneAPI.getProvider().getPrimaryBaritone().getCustomGoalProcess().setGoalAndPath(new GoalXZ((int) mc.player.posX - Integer.MAX_VALUE, (int) mc.player.posZ + Integer.MAX_VALUE));
+
+            direction = "-X / +Z";
+        } else if (normalizeAngle(mc.player.rotationYaw) >= 67.6 && normalizeAngle(mc.player.rotationYaw) <= 112.5) { // -X
+            BaritoneAPI.getProvider().getPrimaryBaritone().getCustomGoalProcess().setGoalAndPath(new GoalXZ((int) mc.player.posX - Integer.MAX_VALUE, (int) mc.player.posZ));
+
+            direction = "-X";
+        } else if (normalizeAngle(mc.player.rotationYaw) >= 112.6 && normalizeAngle(mc.player.rotationYaw) <= 157.5) { // -X / -Z
+            BaritoneAPI.getProvider().getPrimaryBaritone().getCustomGoalProcess().setGoalAndPath(new GoalXZ((int) mc.player.posX - Integer.MAX_VALUE, (int) mc.player.posZ - Integer.MAX_VALUE));
+
+            direction = "-X / -Z";
+        } else if (normalizeAngle(mc.player.rotationYaw) >= 157.6 || normalizeAngle(mc.player.rotationYaw) <= -157.5) { // -Z
+            BaritoneAPI.getProvider().getPrimaryBaritone().getCustomGoalProcess().setGoalAndPath(new GoalXZ((int) mc.player.posX, (int) mc.player.posZ - Integer.MAX_VALUE));
+
+            direction = "-Z";
+        } else if (normalizeAngle(mc.player.rotationYaw) >= -157.6 && normalizeAngle(mc.player.rotationYaw) <= -112.5) { // +X / -Z
+            BaritoneAPI.getProvider().getPrimaryBaritone().getCustomGoalProcess().setGoalAndPath(new GoalXZ((int) mc.player.posX + Integer.MAX_VALUE, (int) mc.player.posZ - Integer.MAX_VALUE));
+
+            direction = "+X / -Z";
+        } else if (normalizeAngle(mc.player.rotationYaw) >= -112.6 && normalizeAngle(mc.player.rotationYaw) <= -67.5) { // +X
+            BaritoneAPI.getProvider().getPrimaryBaritone().getCustomGoalProcess().setGoalAndPath(new GoalXZ((int) mc.player.posX + Integer.MAX_VALUE, (int) mc.player.posZ));
+
+            direction = "+X";
+        } else if (normalizeAngle(mc.player.rotationYaw) >= -67.6 && normalizeAngle(mc.player.rotationYaw) <= -22.6) { // +X / +Z
+            BaritoneAPI.getProvider().getPrimaryBaritone().getCustomGoalProcess().setGoalAndPath(new GoalXZ((int) mc.player.posX + Integer.MAX_VALUE, (int) mc.player.posZ + Integer.MAX_VALUE));
+
+            direction = "+X / +Z";
+        }
+    }
+
+    public String getHudInfo() {
+        return direction;
+    }
+
+    public void onDisable() {
```

Comment:
onEnable you should check if the mode was baritone, and store a boolean, otherwise this won't run if they changed mode while it was enabled for some reason